### PR TITLE
Update CHANGELOG.md

### PR DESCRIPTION
**Pull request by oblm**
#### Description
* Adds a workaround for [IQA-1066](https://improbableio.atlassian.net/browse/IQA-1066).
* Moves the Engine Upgrades up the list of changes as they're the headline changes.

#### Tests
I reproduces this issue and proved that the workaround works.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,10 +6,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased-`x.y.z`] - 2020-xx-xx
 
+### New Known Issues:
+- Some files that were included in Unreal Engine `4.23.x` [are not in](https://forums.unrealengine.com/unreal-engine/announcements-and-releases/1695917-unreal-engine-4-24-released?p=1715142#post1715142) `4.24.3`. This causes errors when you upgrade from `4.23.x` to `4.24.3`. You can resolve this issue by running `git restore .`.
+
 ### Breaking Changes:
 - Simulated Player worker configurations now require a dev auth token and deployment flag instead of a login token and player identity token. See the Example Project for an example of how to set this up.
 
 ### Features:
+- Unreal Engine `4.23.1` is now supported. You can find the `4.23.1` version of our engine fork [here](https://github.com/improbableio/UnrealEngine/tree/4.23-SpatialOSUnrealGDK-preview).
```
Comment (mironec) [inline CHANGELOG.md]: This is also included under 0.8.0-preview, I don't think we need to mention it here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,10 +6,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased-`x.y.z`] - 2020-xx-xx
 
+### New Known Issues:
+- Some files that were included in Unreal Engine `4.23.x` [are not in](https://forums.unrealengine.com/unreal-engine/announcements-and-releases/1695917-unreal-engine-4-24-released?p=1715142#post1715142) `4.24.3`. This causes errors when you upgrade from `4.23.x` to `4.24.3`. You can resolve this issue by running `git restore .`.
```
Comment (mironec) [inline CHANGELOG.md]: I would reword this slightly, to not be misleading.
```suggestion
- After upgrading to 4.24.3 using git, you may be left in a state where several .isph and .ispc files are missing, producing [compile errors](https://forums.unrealengine.com/unreal-engine/announcements-and-releases/1695917-unreal-engine-4-24-released?p=1715142#post1715142). You can fix this by running `git restore .`.
```

Comment (mironec) [reply CHANGELOG.md]: Because I think these files were introduced with 4.24.0, Epic somehow deleted them in 4.24.1 and reintroduced them in 4.24.2, but the upgrade process can miss it for some obscure reason.

**Comment by joshuahuburn** [comment]
This PR will generate merge conflicts with https://github.com/spatialos/UnrealGDK/pull/1923/files where I have fixed up the changelog

File: CHANGELOG.md (hunk 2 above)
Comment (oblm) [reply CHANGELOG.md]: much clearer, thanks

File: CHANGELOG.md (hunk 1 above)
Comment (oblm) [reply CHANGELOG.md]: removed

**Comment by oblm** [comment]
> This PR will generate merge conflicts with https://github.com/spatialos/UnrealGDK/pull/1923/files where I have fixed up the changelog

merge yours first, I'll resolve conflicts after.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,10 +6,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased-`x.y.z`] - 2020-xx-xx
 
+### New Known Issues:
+- After upgrading to Unreal Engine `4.24.3` using `git pull`, you may be left in a state where several `.isph` and `.ispc` files are missinf. This state produces [compile errors](https://forums.unrealengine.com/unreal-engine/announcements-and-releases/1695917-unreal-engine-4-24-released?p=1715142#post1715142) when you build the engine. You can fix this by running `git restore .` in the root of your `UnrealEngine` repository.
```
Comment (mironec) [inline CHANGELOG.md]:
```suggestion
- After upgrading to Unreal Engine `4.24.3` using `git pull`, you may be left in a state where several `.isph` and `.ispc` files are missing. This state produces [compile errors](https://forums.unrealengine.com/unreal-engine/announcements-and-releases/1695917-unreal-engine-4-24-released?p=1715142#post1715142) when you build the engine. You can fix this by running `git restore .` in the root of your `UnrealEngine` repository.
```

**Comment by joshuahuburn** [comment]
0.8.1 has now been merged into master so we should update this PR

**Comment by joshuahuburn** [comment]
@oblm I've fixed the conflicts and re-merged with master. This should be good to go.
